src/genai/services/async_generator.py

```diff
@@ -124,66 +125,76 @@ async def _get_response_json(self, model, inputs, params, options):
+                    batch_num,
+                    len(inputs),
+                    None,
+                    GenAiException("Generation has been aborted by the user."),
                 )
-                return
+            )
+            return
 
-            response = None
-            try:
-                response = await self._get_response_json(self.model_id, inputs, self.params, self.options)
-                logger.debug("Received response = {}".format(response))
-                for i in range(len(response["results"])):
-                    response["results"][i]["input_text"] = inputs[i]
-                response = self.message_type_(**response)
-                logger.debug("Cast to Response = {}".format(response))
-            except Exception as e:
-                logger.error(
-                    "Exception raised async_generate and casting : {}, response = {}, inputs = {}".format(
-                        str(e), response, inputs
-                    )
-                )
-                self.queue_.put_nowait((batch_num, len(inputs), None, to_genai_error(e)))
-                return
-            try:
-                self.queue_.put_nowait((batch_num, len(inputs), response, None))
-                if self.callback is not None:
-                    for result in response.results:
-                        self.callback(result)
-            except Exception as e:
-                logger.error(
-                    "Exception raised in callback : {}, response = {}, inputs = {}".format(str(e), response, inputs)
+        response = None
+        try:
+            response = await self._get_response_json(self.model_id, inputs, self.params, self.options)
+            logger.debug("Received response = {}".format(response))
+            for i in range(len(response["results"])):
+                response["results"][i]["input_text"] = inputs[i]
+            response = self._message_type(**response)
+            logger.debug("Cast to Response = {}".format(response))
+        except Exception as e:
+            logger.error(
+                "Exception raised async_generate and casting : {}, response = {}, inputs = {}".format(
+                    str(e), response, inputs
                 )
+            )
+            self._queue.put_nowait((batch_num, len(inputs), None, to_genai_error(e)))
+            return
+        try:
+            self._queue.put_nowait((batch_num, len(inputs), response, None))
+            if self.callback is not None:
+                for result in response.results:
+                    self.callback(result)
+        except Exception as e:
+            logger.error(
+                "Exception raised in callback : {}, response = {}, inputs = {}".format(str(e), response, inputs)
+            )
 
     async def _schedule_requests(self):
+        async def get_limits():
+            if self.fn == "tokenize":
+                return len(self.prompts)
+
+            limits = self.service.generate_limits()
```

Comment:
I think that this function should be mocked in testing in test_concurrent.py. - test is failing